### PR TITLE
Attempt to close the MMDB handle automatically when it gets deallocated

### PR DESCRIPTION
**Pull request by martinslota**
Not sure at all that this is the way to go - we can discuss.

### CODE REVIEW

File: src/mmdb.ml [hunk 1]
```diff
@@ -96,7 +96,7 @@ module Query_result = struct
 end
 
 let run_query ~lookup_result ~query :
-    (Query_result.t option, Errors.Lookup_error.t) Result.t =
+    (Query_result.t option, Errors.Lookup_error.t) Caml.result =
```
Comment (martinslota) [inline src/mmdb.ml]: Whoops, I'll revert this...

**Comment by Leonidas-from-XIV** [comment]
I would recommend against this, since it mixes the Caml GC with the memory management that ctypes has to do under the hood. ctyles' `allocate` functions have a `finalise` optional argument, I could imagine it would make sense to call `mmdb_close` there instead.

**Comment by martinslota** [comment]
Closes #7

**Comment by martinslota** [comment]
@Leonidas-from-XIV Should be good to go now.

**Pull request merged**